daphne/src/roles_test.rs

```diff
@@ -454,15 +449,11 @@ async fn http_post_aggregate_failure_batch_collected() {
         let mut report_store_mutex_guard = helper.report_store.lock().expect("lock() failed");
         let report_store = report_store_mutex_guard.deref_mut();
 
-        // Create a new instance of report store associated to the bucket_info.
-        let bucket_info = BucketInfo::new(task_config, task_id, &report.nonce);
-        report_store.insert(bucket_info.clone(), ReportStore::new());
-
-        // Intentionally mark report store as `collected`.
-        report_store
-            .get_mut(&bucket_info)
-            .expect("report_store not found")
-            .process_mark_collected();
+        // Create a new instance of report store associated to the task which is intentionally
+        // marked as `collected`.
+        let mut store = ReportStore::new();
+        store.collected = true;
+        report_store.insert(task_id.clone(), store);
     }
```

Comment:
Why is this code in its own block? Could we get rid of this?
```suggestion
```

Reply:
This is necessary because of there is no way to release mutex locks in Rust.